addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -287,17 +288,17 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
                 self._update_product_info(line.product_id, bom.id, product_info, warehouse, line_quantity, bom=False, parent_bom=bom, parent_product=product)
         components_closest_forecasted = self._get_components_closest_forecasted(no_bom_lines, line_quantities, bom, product_info, product, ignore_stock)
         for component_index, line in enumerate(bom.bom_line_ids):
-            new_index = f"{index}{component_index}"
+            new_index = f"{index}/{component_index}"
             if product and line._skip_bom_line(product):
                 continue
             line_quantity = line_quantities.get(line.id, 0.0)
-            if line.child_bom_id:
+            if line.child_bom_id and (max_depth is False or depth < max_depth):
                 component = self._get_bom_data(line.child_bom_id, warehouse, line.product_id, line_quantity, bom_line=line, level=level + 1, parent_bom=bom,
-                                               parent_product=product, index=new_index, product_info=product_info, ignore_stock=ignore_stock)
+                                               parent_product=product, index=new_index, product_info=product_info, ignore_stock=ignore_stock, depth=depth + 1, max_depth=max_depth)
             else:
                 component = self.with_context(
                     components_closest_forecasted=components_closest_forecasted,
-                )._get_component_data(bom, product, warehouse, line, line_quantity, level + 1, new_index, product_info, ignore_stock)
+                )._get_component_data(bom, product, warehouse, line, line_quantity, level + 1, new_index, product_info, bom_id=self._get_component_bom(line.product_id, line.child_bom_id), ignore_stock=ignore_stock)
```

Comment:
This also mess with the fold indexes used in the PDF generation, as the first time a subassembly is loaded, it will have the index `component_[...]` assigned to it. But once you click on it and replace it by the new data, this time it went through `_get_bom_data` and has now an index `bom_[...]`.